ulauncher/ui/windows/PreferencesUlauncherDialog.py

```diff
@@ -276,9 +276,12 @@ def prefs_set_autostart(self, url_params):
 
     @rt.route('/set/show-recent-apps')
     def prefs_set_show_recent_apps(self, url_params):
-        is_on = self._get_bool(url_params['query']['value'])
-        logger.info('Set show-recent-apps to %s', is_on)
-        self.settings.set_property('show-recent-apps', is_on)
+        try:
+            recent_apps_number = int(url_params['query']['value'])
+        except ValueError:
+            recent_apps_number = 3
```

Comment:
I like that you've saved backward compatibility with a boolean setting that existing users have :+1: